src/extension.ts

```diff
@@ -361,523 +368,532 @@ function launchMetals(
-      const editors = window.visibleTextEditors;
-      const path = Uri.parse(params.uri).toString();
-      const editor = editors.find(
-        (editor) => editor.document.uri.toString() == path
+      // NOTE(olafur): `require("./package.json")` should work in theory but it
+      // seems to read a stale version of package.json when I try it.
+      const packageJson = JSON.parse(
+        fs.readFileSync(
+          path.join(context.extensionPath, "package.json"),
+          "utf8"
+        )
       );
-      if (editor) {
-        const options = params.options.map<DecorationOptions>((o) => {
-          return {
-            range: new Range(
-              new Position(o.range.start.line, o.range.start.character),
-              new Position(o.range.end.line, o.range.end.character)
-            ),
-            hoverMessage: o.hoverMessage?.value,
-            renderOptions: o.renderOptions,
-          };
-        });
-        if (params.uri.endsWith(".worksheet.sc"))
-          editor.setDecorations(worksheetDecorationType, options);
-        else editor.setDecorations(decorationType, options);
-      } else {
-        outputChannel.appendLine(
-          `Ignoring decorations for non-active document '${params.uri}'.`
+      const viewIds = packageJson.contributes.views["metals-explorer"].map(
+        (view: { id: string }) => view.id
+      );
+      treeViews = startTreeView(client, outputChannel, context, viewIds);
+      context.subscriptions.concat(treeViews.disposables);
+      scalaDebugger
+        .initialize(outputChannel)
+        .forEach((disposable) => context.subscriptions.push(disposable));
+      client.onNotification(DecorationTypeDidChange.type, (options) => {
+        decorationType = window.createTextEditorDecorationType(options);
+      });
+      client.onNotification(DecorationsRangesDidChange.type, (params) => {
+        const editors = window.visibleTextEditors;
+        const path = Uri.parse(params.uri).toString();
+        const editor = editors.find(
+          (editor) => editor.document.uri.toString() == path
         );
-      }
-    });
-  });
+        if (editor) {
+          const options = params.options.map<DecorationOptions>((o) => {
+            return {
+              range: new Range(
+                new Position(o.range.start.line, o.range.start.character),
+                new Position(o.range.end.line, o.range.end.character)
+              ),
+              hoverMessage: o.hoverMessage?.value,
+              renderOptions: o.renderOptions,
+            };
+          });
+          if (params.uri.endsWith(".worksheet.sc"))
+            editor.setDecorations(worksheetDecorationType, options);
+          else editor.setDecorations(decorationType, options);
+        } else {
+          outputChannel.appendLine(
+            `Ignoring decorations for non-active document '${params.uri}'.`
+          );
+        }
+      });
+    },
+    (err: Error) => {
```

Comment:
if `err` is already of type `Error`, then no need to add the type here:

```suggestion
    (err) => {
```

If not, then we need to check it at runtime (like above)

Reply:
Ach, right. I added a check.